backend/src/folder/folder.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@nestjs/common';
+import { CreateFolderInput } from './dto/create-folder.input';
+import { UpdateFolderInput } from './dto/update-folder.input';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { Folder } from './entities/folder.entity';
+
+@Injectable()
+export class FolderService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async getFolders(userId: string): Promise<Folder[]> {
+    return this.prisma.folder.findMany({
+      where: { userId },
+    });
+  }
+
+  async getFolder(id: string, userId: string): Promise<Folder> {
+    const folder = await this.prisma.folder.findUnique({
+      where: { id, userId },
+      include: { journals: true } as any,
+    });
+
+    if (!folder) {
+      throw new Error('Folder not found or access denied');
+    }
+
+    return folder;
+  }
+
+  async createFolder(
+    input: CreateFolderInput,
+    userId: string,
+  ): Promise<Folder> {
+    const TITLE_MAX_LENGTH = 50;
```

Comment:
You're using TITLE_LENGTH in multiple places

I'd recommend creating a separate const file and putting that TITLE_MAX_LENGTH in there

I would create a file `folder.constants.ts` in your folders module

Reply:
Done!